src/storage/statistics/variant_stats.cpp

```diff
@@ -0,0 +1,181 @@
+	return stats.child_stats[0];
+}
+
+BaseStatistics &VariantStats::GetUnshreddedStats(BaseStatistics &stats) {
+	AssertVariant(stats);
+	return stats.child_stats[0];
+}
+
+void VariantStats::SetUnshreddedStats(BaseStatistics &stats, const BaseStatistics &new_stats) {
+	AssertVariant(stats);
+	stats.child_stats[0].Copy(new_stats);
+}
+
+void VariantStats::SetUnshreddedStats(BaseStatistics &stats, unique_ptr<BaseStatistics> new_stats) {
+	AssertVariant(stats);
+	if (!new_stats) {
+		CreateUnshreddedStats(stats);
+	} else {
+		SetUnshreddedStats(stats, *new_stats);
+	}
+}
+
+void VariantStats::SetShreddedStats(BaseStatistics &stats, const BaseStatistics &new_stats) {
+	AssertVariant(stats);
+	auto &data = GetDataUnsafe(stats);
+	if (!data.is_shredded) {
+		BaseStatistics::Construct(stats.child_stats[1], new_stats.GetType());
+		data.is_shredded = true;
+	}
+	stats.child_stats[1].Copy(new_stats);
+}
+
+void VariantStats::Serialize(const BaseStatistics &stats, Serializer &serializer) {
+	auto &unshredded_stats = VariantStats::GetUnshreddedStats(stats);
+
+	serializer.WriteList(200, "child_stats", 1,
+	                     [&](Serializer::List &list, idx_t i) { list.WriteElement(unshredded_stats); });
+}
+
+void VariantStats::Deserialize(Deserializer &deserializer, BaseStatistics &base) {
+	auto &type = base.GetType();
+	D_ASSERT(type.InternalType() == PhysicalType::STRUCT);
+	D_ASSERT(type.id() == LogicalTypeId::VARIANT);
+
+	auto unshredded_type = VariantShredding::GetUnshreddedType();
+	deserializer.ReadList(200, "child_stats", [&](Deserializer::List &list, idx_t i) {
+		deserializer.Set<const LogicalType &>(unshredded_type);
+		auto stat = list.ReadElement<BaseStatistics>();
+		base.child_stats[i].Copy(stat);
+		deserializer.Unset<LogicalType>();
+	});
+}
+
+string VariantStats::ToString(const BaseStatistics &stats) {
+	auto &data = GetDataUnsafe(stats);
+	return StringUtil::Format("is_shredded: %s", data.is_shredded ? "true" : "false");
+}
+
+void VariantStats::Merge(BaseStatistics &stats, const BaseStatistics &other) {
+	if (other.GetType().id() == LogicalTypeId::VALIDITY) {
+		return;
+	}
+
+	stats.child_stats[0].Merge(other.child_stats[0]);
+	auto &data = GetDataUnsafe(stats);
+	auto &other_data = GetDataUnsafe(other);
+
+	if (other_data.is_shredded) {
+		const bool is_shredded = data.is_shredded;
+		if (!data.is_shredded) {
+			stats.child_stats[1] = BaseStatistics::CreateUnknown(other.child_stats[1].GetType());
+			data.is_shredded = true;
+		}
+		//! FIXME: assumes equal shredding type?
+		stats.child_stats[1].Merge(other.child_stats[1]);
+	}
+}
+
+void VariantStats::Copy(BaseStatistics &stats, const BaseStatistics &other) {
+	stats.child_stats[0].Copy(other.child_stats[0]);
+	auto &data = VariantStats::GetDataUnsafe(stats);
+	auto &other_data = VariantStats::GetDataUnsafe(other);
+	if (other_data.is_shredded) {
+		if (!data.is_shredded) {
+			stats.child_stats[1] = BaseStatistics::CreateUnknown(other.child_stats[1].GetType());
```

Comment:
I think this shouldn't be necessary - we should just copy over the child stats here